src/gbm/gbtree.cc

```diff
@@ -57,6 +57,14 @@ void GBTree::Configure(const Args& cfg) {
 
   monitor_.Init("GBTree");
 
+  if (tparam_.tree_method == TreeMethod::kGPUHist &&
+      std::none_of(cfg.cbegin(), cfg.cend(),
+                   [](std::pair<std::string, std::string> const& arg) {
+                     return arg.first == "predictor";
```

Comment:
Is this check correct? Are you sure this change won't cause the cpu_predictor param to be overwritten?

Reply:
@RAMitchell 

- If "predictor" is specified by user, then it should appear in this configuration and be found by this `std::none_of`, hence nothing changes.
- If `predictor` is loaded from binary model, it should also appear in this configuration, nothing changes.
- If `predictor` is never set, and `tree_method` is not `gpu_hist`, nothing changes.
- If `predictor` is never set, but `tree_method` is `gpu_hist`, then `predictor` **is changed** to `gpu_predictor`.

I most certainly don't like the configuration.  In an ideal world there should be only one `hist` method and only one `predictor`, just like our objective functions and metrics.  We dispatch them by `gpu_id` or by #4600 .  But I don't think we can do that in near future ...

Reply:
My mistake, I thought this was checking the second part of the pair for some reason.